test/e2e/network/netpol/network_policy.go

```diff
@@ -0,0 +1,884 @@
+
+			err := k8s.CleanNetworkPolicies(model.NamespaceNames)
+			time.Sleep(3 * time.Second) // TODO we can remove this eventually, its just a hack to keep CI stable.
+			framework.ExpectNoError(err, "unable to clean network policies")
+
+			// Now the policiy is deleted, we expect all connectivity to work again.
+			reachabilityAll := NewReachability(model.AllPods(), true)
+			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachabilityAll}, isVerbose)
+		})
+
+		ginkgo.It("should allow egress access to server in CIDR block [Feature:Netpol]", func() {
+			// Getting podServer's status to get podServer's IP, to create the CIDR
+			nsX, nsY, _, model, k8s := getK8SModel(f)
+			podList, err := f.ClientSet.CoreV1().Pods(nsY).List(context.TODO(), metav1.ListOptions{LabelSelector: "pod=b"})
+			framework.ExpectNoError(err, "Failing to list pods in namespace y")
+			pod := podList.Items[0]
+
+			hostMask := 32
+			if utilnet.IsIPv6String(pod.Status.PodIP) {
+				hostMask = 128
+			}
+			podServerCIDR := fmt.Sprintf("%s/%d", pod.Status.PodIP, hostMask)
+			policyAllowCIDR := GetAllowEgressByCIDR("a", podServerCIDR)
+			CreateOrUpdatePolicy(k8s, policyAllowCIDR, nsX, true)
+
+			reachability := NewReachability(model.AllPods(), true)
+			reachability.ExpectAllEgress(NewPodString(nsX, "a"), false)
+			reachability.Expect(NewPodString(nsX, "a"), NewPodString(nsY, "b"), true)
+			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachability}, isVerbose)
+		})
+
+		ginkgo.It("should enforce except clause while egress access to server in CIDR block [Feature:Netpol]", func() {
+			// Getting podServer's status to get podServer's IP, to create the CIDR with except clause
+			nsX, _, _, model, k8s := getK8SModel(f)
+			podList, err := f.ClientSet.CoreV1().Pods(nsX).List(context.TODO(), metav1.ListOptions{LabelSelector: "pod=a"})
+			framework.ExpectNoError(err, "Failing to find pod x/a")
+			podA := podList.Items[0]
+
+			podServerAllowCIDR := fmt.Sprintf("%s/4", podA.Status.PodIP)
+
+			podList, err = f.ClientSet.CoreV1().Pods(nsX).List(context.TODO(), metav1.ListOptions{LabelSelector: "pod=b"})
+			framework.ExpectNoError(err, "Failing to find pod x/b")
+			podB := podList.Items[0]
+
+			podServerExceptList := []string{fmt.Sprintf("%s/32", podB.Status.PodIP)}
+			policyAllowCIDR := GetAllowEgressByCIDRExcept("a", podServerAllowCIDR, podServerExceptList)
+			CreateOrUpdatePolicy(k8s, policyAllowCIDR, nsX, true)
+
+			reachability := NewReachability(model.AllPods(), true)
+			reachability.Expect(NewPodString(nsX, "a"), NewPodString(nsX, "b"), false)
+
+			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachability}, isVerbose)
+		})
+
+		ginkgo.It("should ensure an IP overlapping both IPBlock.CIDR and IPBlock.Except is allowed [Feature:Netpol]", func() {
```

Comment:
this seems to work only for IPv4, /4 and /32 masks

Reply:
xref https://github.com/kubernetes/kubernetes/pull/93583

Reply:
ok fixed this one